addonscript/src/addonscript/api.py

```diff
@@ -14,6 +14,36 @@
 UPLOAD_STATUS_PK = "api/v3/addons/{id}/versions/{version}/uploads/{upload_pk}/"
 
 
+# https://addons-server.readthedocs.io/en/latest/topics/api/applications.html
+# XXX when we want to support multiple products, we'll need to unhardcode the
+#     `firefox` below
+ADD_VERSION = "api/v4/applications/firefox/{version}/"
+
+
+async def add_version(context, version):
+    """Add a new version to AMO.
+
+    Use the `min_version` here, rather than the string with the buildid etc.
+
+    Raises:
+        BadVersionError: If the XPI's version fails to sanity check against AMO
```

Comment:
Hm, this isn't accurate. This should be `AuthFailedError` and `AuthInsufficientPermissionsError`.

Reply:
Good catch, fixed!